horace_core/sqw/@sqw/tobyfit_DGdisk_resconv.m

```diff
@@ -141,10 +190,38 @@
 mc_mod_shape_mono = [mc_contributions.moderator,...
     mc_contributions.shape_chopper, mc_contributions.mono_chopper];
 
-% Perform resolution broadening calculation
-% -----------------------------------------
-% Package parameters as a cell for convenience
-if ~iscell(pars), pars={pars}; end
+% Create pointers to parts of lookup structure for the current dataset
+xa=lookup.xa{iw};
+x1=lookup.x1{iw};
+ki=lookup.ki{iw};
+kf = lookup.kf{iw}(idx_start:idx_end, :);
+s_mat=lookup.s_mat{iw};
+spec_to_rlu=lookup.spec_to_rlu{iw};
+alatt=lookup.alatt{iw};
+angdeg=lookup.angdeg{iw};
+is_mosaic=lookup.is_mosaic{iw};
+dt = lookup.dt{iw}(idx_start:idx_end, :);
+en = lookup.en{iw}(idx_start:idx_end, :);
+
+% Run and detector for each pixel
+npix = idx_end - idx_start + 1; %sqw_obj.pix.num_pixels;
+[irun, idet] = parse_pixel_indices(sqw_obj, (idx_start:idx_end)');     % returns column vectors
+
+% Get detector information for each pixel in the sqw object
+% size(x2) = [npix,1], size(d_mat) = [3,3,npix], size(f_mat) = [3,3,npix]
+% and size(detdcn) = [3,npix]
+[x2, detdcn, d_mat, f_mat] = detector_table.func_eval_ind (iw, irun, idet, @detector_info);
+
+% Catch case of refining crystal orientation
+if refine_crystal
```

Comment:
The if refine_crystal ... end block (lines 216 - 224) only need to be called once for any one sqw object, not repeated for every chunk.  Accordingly, the block can be moved to immediately before line 155. If I recall correctly, the call to sample_coords_to_spec_to_rlu is quite time consuming due to the digging down into unique_objects_container &/or unique_reference_container, so this likely will have a practical effect on performance as well as it logically belonging in the main function.